src/samplereader/SubtitleSampleReader.cpp

```diff
@@ -153,7 +153,10 @@ AP4_Result CSubtitleSampleReader::ReadSample()
 void CSubtitleSampleReader::Reset(bool bEOS)
 {
   if (m_adByteStream || bEOS)
+  {
+    m_eos = bEOS;
```

Comment:
`m_eos` not being set was causing the sub reader to be overlooked in `GetNextSample`